### PR TITLE
feat(Toggle group): Add support for icon position

### LINKED ISSUE #10059: ToggleGroup - add iconPosition prop
**Is this a new component or an extension of an existing one?**
Toggle group

**Describe the feature**
Followup to https://github.com/patternfly/patternfly-react/pull/10019
Similar to our Button component, we should add a prop to position the icon in a toggle group item so that it can come before or after the toggle item text (if any). The prop should only accept values "start" and "end" (Button/any other components should have deprecated "left" and "right" values)

**Are there visuals for this feature? If applicable, please include examples for each state and for varying widths**
Include screenshots or links to Marvel or other mockups.

**Any other information?**

### PR DESCRIPTION
**Pull request by tlabaj**
**What**: Closes #10059

### CODE REVIEW

File: packages/react-core/src/components/ToggleGroup/ToggleGroupItem.tsx [hunk 1]
```diff
@@ -53,8 +56,13 @@ export const ToggleGroupItem: React.FunctionComponent<ToggleGroupItemProps> = ({
         disabled={isDisabled}
         id={buttonId}
       >
-        {icon && <ToggleGroupItemElement variant={ToggleGroupItemVariant.icon}>{icon}</ToggleGroupItemElement>}
+        {icon && iconPosition === 'start' && (
+          <ToggleGroupItemElement variant={ToggleGroupItemVariant.icon}>{icon}</ToggleGroupItemElement>
```
Comment (adamviktora) [inline packages/react-core/src/components/ToggleGroup/ToggleGroupItem.tsx:60]: nit: we could extract this ToggleGroupItemElement to a variable to not have it duplicated

**Review by adamviktora — APPROVED** [review]
LGTM

File: packages/react-core/src/components/ToggleGroup/examples/ToggleGroupTextIcon.tsx [hunk 2]
```diff
@@ -15,28 +15,57 @@ export const ToggleGroupIcon: React.FunctionComponent = () => {
...
-      <ToggleGroupItem
-        icon={<ShareSquareIcon />}
-        text="Share"
-        buttonId="toggle-group-text-icons-3"
-        isSelected={isSelected['toggle-group-text-icons-3']}
-        onChange={handleItemClick}
-      />
-    </ToggleGroup>
+    <>
+      <ToggleGroup aria-label="Text and icon toggle group">
```
Comment (thatblindgeye) [inline packages/react-core/src/components/ToggleGroup/examples/ToggleGroupTextIcon.tsx:19]:
```suggestion
      <ToggleGroup aria-label="Text and icon toggle group with icons at start">
```

File: packages/react-core/src/components/ToggleGroup/examples/ToggleGroupTextIcon.tsx [hunk 3]
```diff
@@ -15,28 +15,57 @@ export const ToggleGroupIcon: React.FunctionComponent = () => {
...
+        <ToggleGroupItem
+          icon={<ShareSquareIcon />}
+          text="Share"
+          buttonId="toggle-group-text-icons-3"
+          isSelected={isSelected['toggle-group-text-icons-3']}
+          onChange={handleItemClick}
+        />
+      </ToggleGroup>
+      <br />
+      <ToggleGroup aria-label="Text and icon toggle group">
```
Comment (thatblindgeye) [inline packages/react-core/src/components/ToggleGroup/examples/ToggleGroupTextIcon.tsx:43]:
```suggestion
      <ToggleGroup aria-label="Text and icon toggle group with icons at end">
```

File: packages/react-core/src/components/ToggleGroup/ToggleGroupItem.tsx [hunk 4]
```diff
@@ -8,6 +8,8 @@ export interface ToggleGroupItemProps extends Omit<React.HTMLProps<HTMLDivElemen
   text?: React.ReactNode;
   /** Icon rendered inside the toggle group item */
   icon?: React.ReactNode;
+  /** Sets position of the icon */
```
Comment (thatblindgeye) [inline packages/react-core/src/components/ToggleGroup/ToggleGroupItem.tsx:11]:
```suggestion
  /** Sets position of the icon when text is also passed in */
```

**Review by thatblindgeye — CHANGES_REQUESTED** [review]
A few quick changes below.

Not a blocker, but an example description would be good to add in as well. Maybe something like, "When passing both the `text` and `icon` properties into a `<ToggleGroupItem>`, you can also pass in the `iconPosition` property to determine whether the icon is rendered at the start or end of the item." cc @edonehoo. We could also just open a followup to add descriptions for all ToggleGroup examples since there's not a lot of them, though.

**Comment by edonehoo** [comment]
@thatblindgeye oo yeah let's do a follow up and I can got ahead and fill out the whole page!

**Review by thatblindgeye — APPROVED** [review]

**Pull request merged**